examples/SampleApp/src/screens/NewDirectMessagingScreen.tsx

```diff
@@ -222,24 +226,32 @@ export const NewDirectMessagingScreen: React.FC<NewDirectMessagingScreenProps> =
    *
    * 2. And then navigate to ChannelScreen
    */
-  const customSendMessage = async () => {
-    if (!currentChannel?.current) return;
+  const customSendMessage = async (
+    _: string,
+    message: StreamMessage,
+  ): Promise<SendMessageAPIResponse> => {
+    if (!currentChannel?.current) {
+      throw new Error('Missing current channel');
+    }
 
     if (isDraft.current) {
       currentChannel.current.initialized = false;
       await currentChannel.current.query({});
     }
 
     try {
-      await currentChannel.current.sendMessage({
-        text: messageInputText,
+      const response = await currentChannel.current.sendMessage({
+        text: message.text,
       });
 
       navigation.replace('ChannelScreen', {
         channelId: currentChannel.current.id,
       });
+
```

Comment:
This doesn't cause issues? I would think returning after navigation replacement would cause an attempted state update on an unmounted component?

Reply:
so, after checking this i noticed that I ended up doing that because  `doSendMessageRequest` requires the `ReturnType` of the function to be `Promise<SendMessageAPIResponse<At, Ch, Co, Me, Re, Us>>`

Reply:
Is it not causing a warning though? Was thinking more we should reconsider how this is navigated in the sample app.